definitions/npm/koa-router_v7.2.x/flow_v0.25.x-/koa-router_v7.2.x.js

```diff
@@ -1,20 +1,279 @@
 /**
- * @flow
+ * A lot of type definitions are copied from Koa because we can't `import type` yet
+ * Comments from types removed for brevity. Read the actual `koa` libdef to learn more about these types.
  */
 
-type KoaRouter$Middleware = (
-  ctx: any,
-  next: () => Promise<void>
-) => Promise<void> | void;
+declare module "koa-router" {
+  // Begin `koa` types
+  declare class Server extends net$Server {
+    listen(
+      port?: number,
+      hostname?: string,
+      backlog?: number,
+      callback?: Function
+    ): Server;
+    listen(path: string, callback?: Function): Server;
+    listen(handle: Object, callback?: Function): Server;
+    close(callback?: Function): Server;
+    maxHeadersCount: number;
+    setTimeout(msecs: number, callback: Function): Server;
+    timeout: number;
+  }
 
-type KoaRouter$ParamMiddleware = (
-  param: string,
-  ctx: any,
-  next: () => Promise<void>
-) => Promise<void> | void;
+  declare type JSON = string | number | boolean | null | JSONObject | JSONArray;
+  declare type JSONObject = { [key: string]: JSON };
+  declare type JSONArray = Array<JSON>;
 
-declare module "koa-router" {
-  declare class Router {
+  declare type SimpleHeader = {
```

Comment:
@saadq Would it make sense to prepend all of these imported `koa` types, with something like `Koa$SimpleHeader`? It might help an internal type from one meant to be exported. Would that break anything?

Reply:
Yeah that's a good idea, will do that soon.